src/dlsnode/util/aio/internal/AioScheduler.d

```diff
@@ -59,15 +60,15 @@ class AioScheduler: ISelectEvent
 
     ***************************************************************************/
 
-    private Queue* ready_queue;
+    private TreeQueue!(Job*)* ready_queue;
```

Comment:
`TreeQueue!(Job*) ready_queue` should just work, without `new`ing it in the constructor.

Reply:
These are pointers, used to swap them around.

Reply:
To explain a bit more (I'll document this inline): all the time we have two queues. One (waking up queue) is owned by the main thread, and the other (ready queue) is owned by the thread workers. Whenever we start waking up the fibers running in the main thread, we swap these two queues. Then the thread workers are adding requests that are finished inside `ready_queue`, while the main thread is waking up the sleeping fibers (for which the IO has finished) from the `waking_queue`.

Reply:
Updated with an extra commit with documentation why are these pointers.

Reply:
I see, but still the queues probably don't need to be `new`ed separately but can be embedded in `AioScheduler` like
```d
class AioScheduler
{
    TreeQueue!(Job*)[2] queues;
    TreeQueue!(Job*)* ready_queue, waking_queue;
    this ( )
    {
        this.ready_queue = &this.queues[0];
        this.waking_queue = &this.queues[1];
    }
}
```
if you like.

Reply:
Yeah, that does feel better, I'll update.